src/TArea.cpp

```diff
@@ -40,7 +42,26 @@
 // FIXME: Modify mapper "painter" code to use "exits" rather than deriving the
 // same information each time it is run ???
 
-TArea::TArea(TMap * map , TRoomDB * pRDB )
+static const QColor defaultLabelForeground(QColor(0, 0, 0));
+static const QColor defaultLabelBackground(QColor(0, 0, 0));
+static const int kPixmapDataLineSize = 64;
+
+static const QString scCOLORS{QStringLiteral("colors")};
+static const QString scCOORDINATES{QLatin1String("coordinates")};
+static const QString scGRID_MODE{QStringLiteral("gridMode")};
+static const QString scID{QStringLiteral("id")};
+static const QString scIMAGE{QStringLiteral("image")};
+static const QString scLABELS{QStringLiteral("labels")};
+static const QString scNAME{QStringLiteral("name")};
+static const QString scROOM_COUNT{QStringLiteral("roomCount")};
+static const QString scROOMS{QStringLiteral("rooms")};
+static const QString scSCALED{QStringLiteral("scaled")};
+static const QString scSHOW_ON_TOP{QStringLiteral("showOnTop")};
+static const QString scSIZE{QLatin1String("size")};
+static const QString scTEXT{QStringLiteral("text")};
+static const QString scUSER_DATA{QStringLiteral("userData")};
```

Comment:
These are a pain and make the code difficult to read. Also, since we're creating a QString, there are no advantages to `QStringLiteral` it turns out. I recommend removing these - I understand it's good to have for consistency, but honestly in our small codebase the error is unlikely to happen.

Reply:
Well, we are creating one `QString` (per compilation unit) instead of one for each use within that CU (.cpp file). i will have a look to see whether `QLatin1String` offers a compromise (again there will likely be only one copy per CU as the compiler should share them) so the run-time construction cost should not be that much even with multiple uses through the file.

Reply:
Humm, `QJsonObject` and related has methods that have `QLatin1String` overloads - but they were only introduced in Qt 5.14 - so if I read that right they are reasonable to use given that all our CI builds are using Qt 5.15 and the extra cost of the more complicated construction prior to that is just going to be a burden on systems not at the leading edge of Qt.

Reply:
Sorry, but the developer pain that this code is adding is not worth the convenience it promises. I think if you run the numbers, you won't even see a 5% improvement.